### ISSUE TITLE
Attempt to work around bug in get_type_hints()

### ISSUE THREAD

**Issue by antonagestam** [antonagestam/abcattrs#1]
Currently stumbling hard into this one: https://github.com/python/typing/issues/508

Providing a subclass of `abc.ABC` might be viable.

**Comment by antonagestam** [comment]
Would something like this work?

```python
def dec(cls):
    hints = get_type_hints(cls, local_ns={cls.__name__: cls})
    return cls

@dec
class Foo:
    ref: Foo
```

**Comment by antonagestam** [comment]
As cross-reference, see message here: https://bugs.python.org/issue43463#msg415627

**Comment by antonagestam** [comment]
Solved in #3
